src/devices/bus/gamate/gamate_protection.cpp

```diff
@@ -0,0 +1,113 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+
+#include "emu.h"
+#include "gamate_protection.h"
+
+DEFINE_DEVICE_TYPE(GAMATE_PROT, gamate_protection_device, "gamate_prot", "Gamate Protection Mapper")
+
+gamate_protection_device::gamate_protection_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, GAMATE_PROT, tag, owner, clock),
+	m_is_protection_passed(0),
+	m_has_failed(0),
+	m_passed_write(0),
+	m_inpos(0),
+	m_inbyte(0),
+	m_inseq(0)
+{
+}
+
+void gamate_protection_device::device_start()
+{
+	save_item(NAME(m_is_protection_passed));
+	save_item(NAME(m_has_failed));
+	save_item(NAME(m_passed_write));
+	save_item(NAME(m_inpos));
+	save_item(NAME(m_inbyte));
+	save_item(NAME(m_inseq));
+}
+
+void gamate_protection_device::device_reset()
+{
+	m_is_protection_passed = 0;
+	m_has_failed = 0;
+	m_passed_write = 0;
+	m_inpos = 0;
+	m_inbyte = 0;
+	m_inseq = 0;
+}
+
+bool gamate_protection_device::is_protection_passed()
+{
+	return m_is_protection_passed;
+}
+
+WRITE_LINE_MEMBER(gamate_protection_device::prot_w)
+{
+	logerror("write to protection %01x\n", state);
```

Comment:
Should these `logerror` all be enabled by default, or would it be better to use `logmacro.h` and the `LOG` macro to allow them to be enabled/disabled easily?

Reply:
might make more sense to use the macro, or just remove them now I know it's working as intended